### PR TITLE
Bug: To accept score as final when admin of both playing teams submit a score

### PR DESCRIPTION
**Pull request by Joe2k**

### CODE REVIEW

**Comment by Joe2k** [comment]
Change Log

- Fix the bug where if a admin of both teams in a match enter a score we consider it final and submit it
- If a user is admin of both teams of a match we show both teams spirit scoring form in the same modal to enter 
- Scroll button fix where old js to fetch scroll height was not working correctly hence button is not visible
- Added ability to enter spirit scores including self-scores from the tournament manager

@SibiAkkash @punchagan

**Comment by Joe2k** [comment]
@SibiAkkash Can you review this next? Will merge once reviewed

File: server/api.py [hunk 1]
```diff
@@ -1637,11 +1637,16 @@ def submit_match_score(
...
 
-    if match.status in {Match.Status.COMPLETED, Match.Status.YET_TO_FIX}:
+    if (
+        match.status in {Match.Status.COMPLETED, Match.Status.YET_TO_FIX}
+        or match.team_1 is None
+        or match.team_2 is None
+    ):
         return 400, {"message": "Match score cant be submitted in current status"}
 
-    is_authorised, team_id = can_submit_match_score(match, request.user)
```
Comment (SibiAkkash) [inline server/api.py:1643]: hmm, now that you've replaced the check with `user_tournament_teams`, is the `can_submit_match_score` function being used anywhere else ?

File: frontend/src/components/TournamentMatch.js [hunk 2]
```diff
@@ -833,12 +833,22 @@ const TournamentMatch = props => {
...
+                    />
+                  </Show>
+                  <Show when={isTeamAdminOf(props.match["team_2"].id)}>
+                    <MatchSpiritScoreForm
+                      match={props.match}
+                      tournamentSlug={props.tournamentSlug}
+                      oppTeamNo={1}
+                      curTeamNo={2}
+                    />
+                  </Show>
```
Comment (SibiAkkash) [inline frontend/src/components/TournamentMatch.js]: So if a person is admin of both teams, the spirit score form will be a huge form right ? Would it be better to have 2 spirit-score-submit buttons here, one for each team ?

**Review by SibiAkkash — CHANGES_REQUESTED** [review]

File: server/api.py (hunk 1 above)
Comment (Joe2k) [reply server/api.py:1643]: Yep its not used anywhere

Comment (Joe2k) [reply server/api.py:1643]: I think we can remove the code?

File: frontend/src/components/TournamentMatch.js (hunk 2 above)
Comment (Joe2k) [reply frontend/src/components/TournamentMatch.js]: yeah but then i thought there will be lot of buttons.. and buttons also will be big as we need to mention for which team the spirit button is for

File: server/api.py (hunk 1 above)
Comment (SibiAkkash) [reply server/api.py:1643]: yeah

File: frontend/src/components/TournamentMatch.js (hunk 2 above)
Comment (SibiAkkash) [reply frontend/src/components/TournamentMatch.js]: correct, there might be too many buttons. Yeah and team names can get long too, hmmm

File: server/api.py (hunk 1 above)
Comment (Joe2k) [reply server/api.py:1643]: yee will do

File: frontend/src/components/TournamentMatch.js (hunk 2 above)
Comment (SibiAkkash) [reply frontend/src/components/TournamentMatch.js]: ok, lets leave it like this for now, let's hear admins next tournament if there were any problems.

**Pull request merged**